content/_ext/yaml_gallery_generator.py

```diff
@@ -2,17 +2,95 @@
 from textwrap import dedent
 import pathlib
 
-def build_from_yaml(filename, display_name):
 
-    with open(f'{filename}.yaml') as fid:
-        items = yaml.safe_load(fid)
+def _tag_in_item(item, tag_str):
+    if tag_str is None:
+        return True
+    all_tags = []
+    for k, e in item['tags'].items():
+        all_tags.extend(e)
+    return tag_str in all_tags
+
+
+def _generate_tag_keys(all_items):
+
+    key_set = set()
+    for item in all_items:
+        for k, e in item['tags'].items():
+            key_set.add(k)
+    
+    key_list = list(key_set)
+    key_list.sort()
+
+    return key_list
```

Comment:
The `_sort_tags` functions turns out to do the same thing as the built-in Python function `sorted`.

```suggestion
    return sorted(key_set)
```